src/bitmessageqt/settings.py

```diff
@@ -342,6 +342,8 @@ def accept(self):
         if proxytype_index == 0:
             if self._proxy_type and state.statusIconColor != 'red':
                 self.net_restart_needed = True
+        elif state.statusIconColor == 'red' and self.config.safeGetBoolean('bitmessagesettings', 'dontconnect'):
```

Comment:
I would drop the color test, and put the `dontconnect` test as the first of the tests (if, not elif).

Reply:
I'm not sure the bug is here. What about test? I think I had a fix for #1656 issue when it was reported. But I preferred first to  write a test in order to prevent reintroducing it.